### PR TITLE
Session, proxyInfo and CustomData are passed to prepareRequestFunction (cheerio scraper)

### PR DESCRIPTION
**Pull request by jbartadev**
Added:

-  `customData` 

- `session`

- `proxyInfo`

to `prepareRequestFunction` in cheerio scraper

### CODE REVIEW

File: cheerio-scraper/src/crawler_setup.js [hunk 1]
```diff
@@ -201,7 +205,10 @@ class CrawlerSetup {
 
         if (this.evaledPrepareRequestFunction) {
             try {
-                await this.evaledPrepareRequestFunction({ request, Apify });
+                const { customData } = this.input;
+                const sessionId = session ? session.id : undefined;
+                const proxyInfo = this.proxyConfiguration.newProxyInfo(sessionId);
+                await this.evaledPrepareRequestFunction({ request, session, proxyInfo, Apify, customData });
```
Comment (mnmkng) [inline cheerio-scraper/src/crawler_setup.js:211]: `prepareRequestFunction` gets `proxyInfo` as a parameter so no need to do this here. Just add it on line 188 and pass it here

Comment (mnmkng) [reply cheerio-scraper/src/crawler_setup.js:211]: https://sdk.apify.com/docs/typedefs/prepare-request-inputs#docsNav

**Review by mnmkng — CHANGES_REQUESTED** [review]

File: cheerio-scraper/src/crawler_setup.js (hunk 1 above)
Comment (mnmkng) [reply cheerio-scraper/src/crawler_setup.js:211]: Also means that all the above changes are not needed.

Comment (jbartadev) [reply cheerio-scraper/src/crawler_setup.js]: Ahh, I see. Changed.

**Review by mnmkng — APPROVED** [review]
Cool, I thought the `null` `proxyConfiguration` will need to be removed too, but `Apify.createProxyConfiguration()` will return `undefined` so it's fine and should not cause problems in 0.22 when we switch to `ow` validation.

**Pull request merged**